server/rtcd.go

```diff
@@ -348,77 +351,55 @@ func (m *rtcdClientManager) newRTCDClient(rtcdURL, host string, dialFn rtcd.Dial
 	// Remove trailing slash if present.
 	rtcdURL = strings.TrimSuffix(rtcdURL, "/")
 
-	clientCfg, err := m.getRTCDClientConfig(rtcdURL, dialFn)
+	clientCfg, err := m.getRTCDClientConfig(rtcdURL)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get rtcd client config: %w", err)
 	}
 
-	registerClient := func() error {
-		mutex, err := cluster.NewMutex(m.ctx.API, "rtcd_registration")
-		if err != nil {
-			return fmt.Errorf("failed to create cluster mutex: %w", err)
-		}
-
-		lockCtx, cancelCtx := context.WithTimeout(context.Background(), lockTimeout)
-		defer cancelCtx()
-		if err := mutex.LockWithContext(lockCtx); err != nil {
-			return fmt.Errorf("failed to acquire cluster lock: %w", err)
-		}
-		defer mutex.Unlock()
-
-		if err := m.registerRTCDClient(clientCfg, dialFn); err != nil {
-			return fmt.Errorf("failed to register rtcd client: %w", err)
-		}
-
-		return nil
-	}
-
-	reconnectCb := func(c *rtcd.Client, attempt int) error {
+	var reconnectCb rtcd.ClientReconnectCb
+	reconnectCb = func(c *rtcd.Client, attempt int) error {
```

Comment:
With this PR we are essentially treating this as a simple disconnected callback and dealing with the actual reconnection on the plugin side. This simplifies things as we have more context to operate on but the naming here is very misleading so I am potentially looking to change this or remove it entirely from the `rtcd` side to avoid further confusion.

Reply:
I'm not sure what you mean by `the naming here is very misleading` -- at first I was going to suggest renaming it to `disconnectCb`, but looking at the code, this is the reconnect handler... so what's the misleading part?

Reply:
Mostly the fact that the reconnection would happen after the callback if we returned a nil error. But the way we are using it now is purely to detect disconnection since we create a brand new client and return a fake error just to exit the previous handler.